library/core/src/option.rs

```diff
@@ -159,11 +159,24 @@
 //!
 //! ## Querying the variant
 //!
-//! The [`is_some`] and [`is_none`] methods return [`true`] if the [`Option`]
-//! is [`Some`] or [`None`], respectively.
+//! The [`is_some`] and [`is_none`] methods take a borrow of the [`Option`]
+//! and return [`true`] if the [`Option`] is [`Some`] or [`None`], respectively.
+//!
+//! The [`is_some_and`] and [`is_none_or`] methods take ownership of the [`Option`]
+//! and apply the provided function to make a decision.
+//! The methods return the same boolean value as the function returns.
 //!
 //! [`is_none`]: Option::is_none
 //! [`is_some`]: Option::is_some
+//! [`is_some_and`]: Option::is_some_and
+//! [`is_none_or`]: Option::is_none_or
+//!
+//! ## Inspecting the variant
+//!
+//! The [`inspect`] method takes ownership of the [`Option`]
+//! and applies the provided function to the contained value by reference if [`Some`]
```

Comment:
This should be put under the "These methods transform the [`Some`] variant:" section (which should be changed to "transform or inspect") since it fits better there.